hadoop-cloud/pom.xml

```diff
@@ -135,6 +135,18 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>com.google.cloud.bigdataoss</groupId>
+      <artifactId>gcs-connector</artifactId>
+      <version>${gcs-connector.version}</version>
+      <classifier>shaded</classifier>
+      <exclusions>
+        <exclusion>
+          <groupId>*</groupId>
```

Comment:
curious why do we exclude everything from the shaded jar

Reply:
Thank you for review, @sunchao . According to the shaded pattern,

https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/8453ce7ce7510e983bae7470909fbd02704c0539/gcs/pom.xml#L208-L363

We have all we needed for Hadoop3 and Hadoop2.
- For Hadoop3, https://mvnrepository.com/artifact/com.google.cloud.bigdataoss/gcs-connector/hadoop3-2.2.7
- For Hadoop2, https://mvnrepository.com/artifact/com.google.cloud.bigdataoss/gcs-connector/hadoop2-2.2.7

I intentionally exclude everything. We will add Spark's version if there is additional missing transitive dependency (if exists).

Reply:
issue is that there's a history of the shaded connector still declaring a dependence on things which are now shaded, so breaking convergence.